src/feat/online-feature.cc

```diff
@@ -24,82 +24,91 @@
 
 namespace kaldi {
 
-
 template<class C>
 void OnlineGenericBaseFeature<C>::GetFrame(int32 frame,
                                            VectorBase<BaseFloat> *feat) {
-  KALDI_ASSERT(frame >= 0 && frame < num_frames_);
-  KALDI_ASSERT(feat->Dim() == Dim());
-  feat->CopyFromVec(features_.Row(frame));
+  // 'at' does size checking.
+  feat->CopyFromVec(*(features_.at(frame)));
 };
 
 template<class C>
-bool OnlineGenericBaseFeature<C>::IsLastFrame(int32 frame) const {
-  return (frame == num_frames_ - 1 && input_finished_);
-}
-
-template<class C>
 OnlineGenericBaseFeature<C>::OnlineGenericBaseFeature(
-    const typename C::Options &opts)
-    :mfcc_or_plp_(opts), input_finished_(false), num_frames_(0),
-    sampling_frequency_(opts.frame_opts.samp_freq) { }
+    const typename C::Options &opts):
+    computer_(opts), window_function_(computer_.GetFrameOptions()),
+    input_finished_(false), waveform_offset_(0) { }
 
 template<class C>
 void OnlineGenericBaseFeature<C>::AcceptWaveform(BaseFloat sampling_rate,
-                                        const VectorBase<BaseFloat> &waveform) {
-  if (waveform.Dim() == 0) {
-    return;  // Nothing to do.
-  }
-  if (input_finished_) {
-    KALDI_ERR << "AcceptWaveform called after InputFinished() was called.";
-  }
-  if (sampling_rate != sampling_frequency_) {
-    KALDI_ERR << "Sampling frequency mismatch, expected "
-              << sampling_frequency_ << ", got " << sampling_rate;
-  }
-
-  Vector<BaseFloat> appended_wave;
-
-  const VectorBase<BaseFloat> &wave_to_use = (waveform_remainder_.Dim() != 0 ?
-                                              appended_wave : waveform);
-  if (waveform_remainder_.Dim() != 0) {
-    appended_wave.Resize(waveform_remainder_.Dim() +
-                         waveform.Dim());
-    appended_wave.Range(0, waveform_remainder_.Dim()).CopyFromVec(
-        waveform_remainder_);
-    appended_wave.Range(waveform_remainder_.Dim(),
-                        waveform.Dim()).CopyFromVec(waveform);
-  }
-  waveform_remainder_.Resize(0);
-
-  Matrix<BaseFloat> feats;
-  BaseFloat vtln_warp = 1.0;  // We don't support VTLN warping in this wrapper.
-  mfcc_or_plp_.Compute(wave_to_use, vtln_warp, &feats, &waveform_remainder_);
+                                                 const VectorBase<BaseFloat> &waveform) {
```

Comment:
Either GitHub shows it weirdly, or the whole body of the function is indented by 1 space.

Regarding the braces around a single-statement body, I find compacting code vertically more aesthetic (assume you do too, judging by this change), but I am reminded of a [horror story of the Apple SSL security bug](https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/). Someone duplicated a line by mistake, ending up with this pattern:
```c
if ((err = security_check_1())
  goto fail;
if ((err = security_check_2())
  goto fail;
  goto fail;
err = security_check_3();
fail:
  return err;
```
Easy to slip past quick eyeballing! This went past a peer review with flying colors and was widely deployed over the world... After that was discovered, many teams at Google mandated the use of `{ }` around a single-statement block (and the one I worked in had that rule before I joined). I am not personally excited with using braces everywhere, but that story convinced the rational part of me to adopt that practice.